src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,159 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    public MyLinkedList() {
+        head = null;
+        tail = null;
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(tail, value, null);
```

Comment:
When creating a new `Node<T>`, pass all the fields to the `Node` constructor. This is already done correctly, but ensure that the `Node` class does not have getters and setters as we have access to private fields of the inner class.